runtime/mandala/src/benchmarking/nutsfinance_stable_asset.rs

```diff
@@ -78,49 +92,45 @@ runtime_benchmarks! {
 		let mint_fee = 10000000u128;
 		let swap_fee = 20000000u128;
 		let redeem_fee = 50000000u128;
-		let intial_a = 100u128;
+		let intial_a = 10000u128;
 		let fee_recipient: AccountId = account("fee", 0, SEED);
-		let _ = StableAsset::create_pool(RawOrigin::Root.into(), pool_asset, assets, precisions, mint_fee, swap_fee, redeem_fee, intial_a, fee_recipient.clone());
+		let yield_recipient: AccountId = account("yield", 1, SEED);
+		let _ = StableAsset::create_pool(RawOrigin::Root.into(), pool_asset, assets, precisions, mint_fee, swap_fee, redeem_fee, intial_a, fee_recipient, yield_recipient, 1000000000000000000u128);
 		let pool_id = StableAsset::pool_count() - 1;
 		set_balance_fungibles(LDOT, &tester, 200000000000u128);
 		set_balance_fungibles(AUSD, &tester, 200000000000u128);
 		let _ = StableAsset::mint(RawOrigin::Signed(tester.clone()).into(), pool_id, vec![10000000000u128, 20000000000u128], 0u128);
 	}: _(RawOrigin::Signed(tester), pool_id, 0, 1, 5000000u128, 0u128)
 
```

Comment:
swap's weight is a function of the length of `pool_id.balances`. It calls into `update_balance` which reads into storage for each iteration of balances. 
``` rust
fn update_balance(
		pool_info: &mut StableAssetPoolInfo<
			Self::AssetId,
			Self::AtLeast64BitUnsigned,
			Self::Balance,
			Self::AccountId,
			Self::BlockNumber,
		>,
	) -> DispatchResult {
		for (i, balance) in pool_info.balances.iter_mut().enumerate() {
			let balance_of: Self::AtLeast64BitUnsigned =
				T::Assets::balance(pool_info.assets[i], &pool_info.account_id).into();
			*balance = balance_of
				.checked_mul(&pool_info.precisions[i])
				.ok_or(Error::<T>::Math)?
				.into();
		}
		Ok(())
	}
```

Reply:
fixed